src/posts/2020-09-19-how-to-write-accessible-forms.md

```diff
@@ -0,0 +1,158 @@
+
+Here are some  practical tips on how to write accessible forms.
+
+
+## Always label your forms
+
+Labels are a great way to get started with writing accessible forms. They are vital for helping people who use [screen readers](https://en.wikipedia.org/wiki/Screen_reader) and other kinds of assistive technology determine what label text go with what input field.
+
+There are two ways to associate a `<label>` with an `<input>`.
+- Set the identifier(ID) in the `<input>` to correspond with the `for` attribute of the `<label>`.
+```html
+<form>
+  <label for="firstName">First Name</label>
+  <input type="text" id="firstName" name="firtName"/>
+</form>
+```
+
+- Nest `<input>` directly inside `<label>`.
+```html
+<form>
+  <label>
+    First Name
+    <input type="text" name="firstName">
+  </label>
+</form>
+```
+
+**Note**: The first method is the preffered approach to associate a label with a form control . Additionally, some assistive devices (such as [Dragon NaturallySpeaking](https://en.wikipedia.org/wiki/Dragon_NaturallySpeaking)) do not support the latter.
+
+
+## Highlight input element on focus
+
+Form insertion [carets](https://developer.mozilla.org/en-US/docs/Glossary/caret#:~:text=A%20caret%20sometimes%20called%20a,is%20called%20the%20insertion%20point.) can be difficult to track. A simple solution is to highlight `<input>` on [focus](https://developer.mozilla.org/en-US/docs/Web/CSS/:focus). This way, people do not have to guess where they are on the form. 
+
+![A form with an input field highlighted in blue.](../img/posts/2020-09-19-how-to-write-accessible-forms/form-field-on-focus.png)
+
+```css
+input:focus{
+  outline: 2px solid royalblue;
+  box-shadow: 1px 1px 8px 1px royalblue;
+}
+```
+
+## Break long forms into smaller sections
+
+Filling long forms can be stressful. This is largely true for almost everyone but particularity for people  with [AD/HD](https://www.cdc.gov/ncbddd/adhd/facts.html). Such forms can lead to an increase in [Bounce rate](https://en.wikipedia.org/wiki/Bounce_rate) which isn't what anyone wants. A simple solution is to break them into smaller sections. 
+
+This helps make long forms less daunting and easier to understand as it is provided in bits.
+
+
+The following principles apply for multi-step forms:
+- Repeat overall instruction on every page.
+- The forms should be split into logical groups.
+- The forms should inform the person about the progress thay are making.
+
+
+## Provide error messages if any
+
+If your site has forms, validation messaging and errors are inevitable.
+
+When people encounter errors, clearly they need more help. Some people, like the visually impaired, may not even be aware that an error has occurred. It is common practice to change `<input>` color to red which is not ideal. This is because assistive devices will not catch the error, hence not informing the person of the error.

```

Comment:
I think this paragraph could be flipped so it reads:

1. Common practice is to use only a red color for inputs
2. Not everyone can see or understand the significance of that
3. Error messages help those people out
4. Error messages should be helpful when shown